bazel/dependencies/libz.BUILD.bazel

```diff
@@ -0,0 +1,41 @@
+load("@rules_foreign_cc//foreign_cc:defs.bzl", "cmake")
+
+filegroup(
+    name = "libz_all",
+    srcs = glob(["**"]),
+    visibility = ["//visibility:public"],
+)
+
+filegroup(
+    name = "libz_headers",
+    srcs = glob(["**/*.h"]),
+    visibility = ["//visibility:public"],
+)
+
+
+# zlib
+# The name of this rule is libz to avoid conflicts with the system `@zlib//:zlib`.
+# We cannot use @zlib//:zlib because the libraries that are generated are named
+# libzlib.(a|so), which is incompatible with other foreign_cc builds.
+cmake(
+    name = "libz", 
```

Comment:
I'd prefer to go with a bazel native build of zlib (and libpng for that matter) - then using the foreign_cc rules.

Reply:
I understand the pros of this preference. However, I think that is quite a lot of work without a driving need. My feeling on this is:

1. We are already using foreign_cc rules for other dependency
2. It is a lot of work to completely recreate all the configure logic for every third party dependency

I am happy to go in that direction if there is a consensus that we need to change our approach for dependencies moving forward.